### PR TITLE
Accept MAV_CMD_NAV_WAYPOINT with COMMAND_LONG and COMMAND_INT in GUIDED mode

### PR DESCRIPTION
**Pull request by mustafa-gokce**

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -723,6 +723,7 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_packet(const mavlink_command_i
         return MAV_RESULT_UNSUPPORTED;
 
     case MAV_CMD_DO_REPOSITION:
+    case MAV_CMD_NAV_WAYPOINT:
```
Comment (amilcarlucas) [inline ArduCopter/GCS_Mavlink.cpp:726]: Indentation

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -4770,6 +4770,7 @@ bool GCS_MAVLINK::command_long_stores_location(const MAV_CMD command)
     case MAV_CMD_DO_SET_ROI_LOCATION:
     case MAV_CMD_NAV_TAKEOFF:
     case MAV_CMD_DO_REPOSITION:
+    case MAV_CMD_NAV_WAYPOINT:
```
Comment (amilcarlucas) [inline libraries/GCS_MAVLink/GCS_Common.cpp:4773]: Indentation

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: ArduCopter/GCS_Mavlink.cpp [hunk 3]
```diff
@@ -1009,6 +1010,14 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
         return handle_command_pause_continue(packet_int);
     }
+
+    // waypoint navigation
+    case MAV_CMD_NAV_WAYPOINT: {
+        mavlink_command_int_t packet_int;
+        GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
+        return handle_command_int_do_reposition(packet_int);
```
Comment (peterbarker) [inline ArduCopter/GCS_Mavlink.cpp:1018]: Isn't the function you're calling here going to ascribe special meaning to param2 which `NAV_WAYPOINT` doesn't have, or has different meanings for?

**Review by WickedShell — CHANGES_REQUESTED** [review]
Why support `COMMAND_LONG` at all for this? Anything that uses a position for lat/lon is going to experience a fairly significant loss in accuracy. In general anything that could be used to ingest a position like this we should be trying to not add any new paths that are a guaranteed low input accuracy.

I'm also wondering what the value is of using `NAV_WAYPOINT` instead of `DO_REPOSITION` ? As @peterbarker notes theres differences from the spec for what should happen with param2.

**Comment by mustafa-gokce** [comment]
@amilcarlucas @peterbarker @WickedShell with this update, I tried to implement this without changing any parameters of both commands.
I think this version should supports param2 (and others) now.
Also added protections for zero fields.

**Comment by WickedShell** [comment]
What is the value/advantage of using this over `MAV_CMD_DO_REPOSITION` which we already support?

I'm just struggling at the moment to see what this enables that we were missing before.

**Review by khancyr — CHANGES_REQUESTED** [review]
I don't understand why use this kind of message on copter , specially when we already have a lot of message to so. This is looks an old hack from when the setpoints weren't things.

Is there some good advantage to use those messages as commands ?

**Comment by amilcarlucas** [comment]
@peterbarker didn't you implement this already in 4.5.0? I think this can be closed.

**Comment by peterbarker** [comment]
@amilcarlucas No, that chunk of work was wround commands rather than nav-waypoint-in-guided mode.

'though I most definitely do want to kill nav-waypoint-in-guided-mode!  A cominbation of `MAV_CMD_DO_REPOSITION`, fixing Plane's change-altitude stuff to solely use set-position-target, and for Copter something to do with changing yaw should allow us to kill the abomination.

Thanks anyway, @mustafa-gokce .

**Comment by amilcarlucas** [comment]
Should this be reopened and rebased then?

**Pull request closed without merging**